web/docs/understand-vast/data-model/taxonomies.md

```diff
@@ -0,0 +1,215 @@
+embarking on a long journey, we have to appreciate that it will always remain an
+imperfect approximation in practice, for the following reasons:
+
+- **Incompleteness**: we have to appreciate that all data models are incomplete
+  because data sources continuously evolve.
+- **Incorrectness**: in addition to lacking information, data models contain
+  a growing number errors, for the same evolutionary reasons as above.
+- **Variance**: data models vary substantially between products, making it
+  difficult to mix-and-match semantics.
+:::
+
+## Concepts
+
+A *concept* is a set of [extractors][extractors] to enable more semantic
+querying. VAST translates a query expression containing a concept to a
+disjunction of all extractors.
+
+[extractors]: /docs/understand-vast/query-language/expressions#extractors
+
+For example, Consider Sysmon and Suricata events, each of which have a notion of
+a network connection with a source IP address. The Sysmon event
+`NetworkConnection` contains a field `SourceIp` and the Suricata event `flow`
+contains a field `src_ip` for this purpose. Without concepts, querying for a
+specific value would involve writing a disjunction of two predicates:
+
+```c
+suricata.flow.src_ip == 6.6.6.6 || sysmon.NetworkConnection.SourceIp == 6.6.6.6
+```
+
+With concepts, you can write this as:
+
+```c
+source_ip == 6.6.6.6
+```
+
+Concepts decouple semantics from syntax and allow you to write queries that
+"scale" independent of the number of data sources. No one wants to remember
+all format-specific names, aside from being an error-prone practice.
+
+You can define a concept in a [module](modules) as follows:
+
+```yaml
+concepts:
+  source_ip:
+    description: the originator of a network-layer connection
+    fields:
+    - sysmon.NetworkConnection.SourceIp
+    - suricata.flow.src_ip
+```
+
+Concepts have no explicit type because extractors have no type either. Queries
+remain meaningful even if types differ across extractors, because VAST ignores
+all fields where the type does not match. For example, if you define a concept
+`x` that includes two fields `a` of type `real` and `b` of type `string`, then
+the query `x < 4.2` would only consider field `a`. In this regard, you can think
+of concepts like [sum types](https://en.wikipedia.org/wiki/Sum_type).
+
+Concepts compose. A concept can include other concepts to represent semantic
+hierarchies. For example, consider our above `source_ip` concept. If we want to
+generalize this concept to also include MAC addresses, we could define a concept
+`source` that includes both `source_ip` and a new field that represents a MAC
+address:
+
+![Concept Composition](/img/concept-composition.light.png#gh-light-mode-only)
+![Concept Composition](/img/concept-composition.dark.png#gh-dark-mode-only)
+
+You define the composite concept in a module as follows:
+
+```yaml
+concepts:
+  source_ip:
+    description: the originator of a connection
+    fields: # TODO: extractors?
```

Comment:
I put this out there for @dominiklohmann's extractor changes. Because concepts will resolve to more than just fields, I wonder whether we need to changes this.

For models, we use the key `definition` already, so we could use that here as well.

Reply:
I think we need to do a more general rewording pass for some options in that regard, similar to the "event type" vs. "type" vs. "schema" vs. "layout" mix we currently still have. We have a story for this already in the catalog epic, but for now it makes more sense to document the status quo here.